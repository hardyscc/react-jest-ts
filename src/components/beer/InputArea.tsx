import React from "react";

interface IProps {
  onSubmit: (name: string) => void;
}

interface IState {
  text: string;
}

class InputArea extends React.Component<IProps, IState> {
  state: IState = {
    text: ""
  };

  render() {
    return (
      <div>
        <input value={this.state.text} onChange={this.setText} />
        <button onClick={this.handleClick}>Add</button>
      </div>
    );
  }

  setText = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({ text: event.target.value });
  };

  handleClick = () => {
    if (this.state.text) {
      this.props.onSubmit(this.state.text);
      this.setState({ text: "" });
    }
  };
}

export default InputArea;
