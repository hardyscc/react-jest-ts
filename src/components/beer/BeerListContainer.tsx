import React from "react";
import BeerList from "./BeerList";
import InputArea from "./InputArea";

interface IProps {
  title?: string;
}

interface IState {
  beers: string[];
}

class BeerListContainer extends React.Component<IProps, IState> {
  static defaultProps: Partial<IProps> = {
    title: "Beer List"
  };

  state: IState = {
    beers: []
  };

  render() {
    return (
      <React.Fragment>
        <InputArea onSubmit={this.addItem} />
        <BeerList items={this.state.beers} />
      </React.Fragment>
    );
  }

  addItem = (name: string) => {
    this.setState({
      beers: [...this.state.beers, name]
    });
  };
}

export default BeerListContainer;
