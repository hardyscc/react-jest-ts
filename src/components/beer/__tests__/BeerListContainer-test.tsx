import { mount, shallow } from "enzyme";
import React from "react";
import BeerListContainer from "../BeerListContainer";
import InputArea from "../InputArea";

describe("BeerListContainer", () => {
  it("should render InputArea and BeerList", () => {
    const wrapper = shallow(<BeerListContainer />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should start with an empty list", () => {
    const wrapper = shallow(<BeerListContainer />);
    expect(wrapper.state("beers")).toEqual([]);
  });

  it("adds items to the list", () => {
    const wrapper = shallow(<BeerListContainer />);
    const container = wrapper.instance() as BeerListContainer;
    container.addItem("Sam Adams");
    expect(wrapper.state("beers")).toEqual(["Sam Adams"]);
  });

  it("passes addItem to InputArea", () => {
    const wrapper = shallow(<BeerListContainer />);
    const inputArea = wrapper.find(InputArea);
    const container = wrapper.instance() as BeerListContainer;
    expect(inputArea.prop("onSubmit")).toEqual(container.addItem);
  });

  it("passes a bound addItem function to InputArea", () => {
    const wrapper = shallow(<BeerListContainer />);
    const inputArea = wrapper.find(InputArea);
    inputArea.prop("onSubmit")("Sam Adams");
    expect(wrapper.state("beers")).toEqual(["Sam Adams"]);
  });

  it("renders the items", () => {
    const wrapper = mount(<BeerListContainer />);
    const container = wrapper.instance() as BeerListContainer;
    container.addItem("Sam Adams");
    container.addItem("Resin");
    wrapper.update();
    expect(wrapper.find("li")).toHaveLength(2);
  });
});
