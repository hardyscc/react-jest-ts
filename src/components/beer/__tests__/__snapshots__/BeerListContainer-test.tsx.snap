// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerListContainer should render InputArea and BeerList 1`] = `
<Fragment>
  <InputArea
    onSubmit={[Function]}
  />
  <BeerList
    items={Array []}
  />
</Fragment>
`;
